### PR TITLE
fix(DB/Netherstorm): adding CC immunities to Ethereum Prisoners

### PR DESCRIPTION
**Pull request by damnablebear**
fix(DB/Netherstorm): adding CC immunities to Ethereum Prisoners

## Changes Proposed:
This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

If your pull request promotes complex changes that require a detailed explanation, please describe them in detail specifying what your solution is and what is it meant to address.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/15395
- Closes https://github.com/chromiecraft/chromiecraft/issues/5178

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [x] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [x] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

Steps to test anyway:

Creature IDs: 22827, 22826, 22825, 20888, 22828, 22920
- .go c id  for any of the above creature ids
- try using any stuns, roots, fears, etc. They should only be able to be interrupted.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by proxical** [comment]
Tested and confirmed to be working as is intended in all scenarios with very minor unexpected behavior as is specified below.
All characters used for testing were freshly created, set to level 80 via .character level, given all class spells and talents, and given the QA Debug Instant Cast Buff from the Blizzard Internal tree when necessary.
The listed NPCs were spawned in an otherwise deserted area of Durotar, before having the abilities listed below used on them. An analysis was conducted to determine the result of using each ability.
The following CC abilities were used and had no effect:
Hunter: Freezing trap (caused the NPCs to visually recoil slightly and released a green cloud, but had no other impact) and Wyvern Sting (did not trigger at all and did not in any way affect the spawned NPCs)
Warlock: Fear w/ Glyph of Fear (returned immune damage tag) and Banish, which showed immune on all except 22920 and 22827 which returned invalid target
Druid: Entangling Roots (returned immune damage tag), Hibernate (returned invalid target on all), Cyclone (returned immune damage tag), Bash in bear form (returned immune damage tag), and Pounce from cat form (caused NPCs to become red with the standard blood effect, however did not block and/or impede movement at all)
Priest: Shackle Undead (returned invalid target) and Mind Control (returned invalid target on all except 22826 which returned immune damage tag)
Paladin: Repentance (returned immune damage tag except for on 22828 where it returned invalid target)
Shaman: Hex (returned invalid target except for on 22826)
Mage: Polymorph (returned invalid target except for on 22826)
As a result, it can conclusively be determined that the patch successfully made all six listed NPCs immune to all CC abilities (listed above). The patch works as intended, and to the best of my knowledge after extensive testing, does not cause any unintended behaviors or issues which did not already occur before this patch. Confirmed to be working.

File: data/sql/updates/pending_db_world/ethereum-prisoner-update.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+UPDATE `creature_template` SET `mechanic_immune_mask` = `mechanic_immune_mask`|1|2|8|16|32|64|256|512|1024|2048|4096|8192|16384|65536|131072|262144|524288|2097152|4194304|67108864|536870912|1073741824 WHERE `entry` IN (22827, 22826, 22825, 20888, 22828, 22920);
```
Comment (elthehablo) [inline data/sql/updates/pending_db_world/ethereum-prisoner-update.sql:1]:
```suggestion
UPDATE `creature_template` SET `mechanic_immune_mask` = `mechanic_immune_mask`|1|2|8|16|32|64|256|512|1024|2048|4096|8192|65536|131072|524288|4194304|8388608|67108864|536870912 WHERE `entry` IN (22827, 22826, 22825, 20888, 22828, 22920);
```

add horror
remove enrage, infected, shield, bleed

**Review by elthehablo — CHANGES_REQUESTED** [review]
these are the ones that should be on the creature according to me, feel free to discuss

**Comment by damnablebear** [comment]
I've committed @elthehablo 's changes and this is ready for re-review.

**Review by elthehablo — APPROVED** [review]

**Comment by Gultask** [comment]
Remember to add the chromiecraft issue as well :)

**Comment by damnablebear** [comment]
> Remember to add the chromiecraft issue as well :)

Isn't it added above, in the Issues Addressed section? Or is there somewhere else to add it?

**Comment by elthehablo** [comment]
> > Remember to add the chromiecraft issue as well :)
> 
> Isn't it added above, in the Issues Addressed section? Or is there somewhere else to add it?

It is, as far as I can see

**Comment by Tereneckla** [comment]
> > > Remember to add the chromiecraft issue as well :)
> > 
> > 
> > Isn't it added above, in the Issues Addressed section? Or is there somewhere else to add it?
> 
> It is, as far as I can see

looks like Gultask added it

**Review by Gultask — APPROVED** [review]

**Pull request merged**